Assets/Scripts/Controllers/EnemyController.cs

```diff
@@ -119,6 +127,8 @@ public Enemy GetEnemyType(string enemyType = null)
                     break;
             }
 
+            if (newEnemy != null) newEnemy.EcRef = this;
```

Comment:
When instantiating a new enemy we ensure it has a reference to the controller that created it